// react-bulma-companion/lib/Image/Image.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Image;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _ImageContent = _interopRequireDefault(require("./ImageContent"));

var _excluded = ["children", "className", "component", "fullwidth", "ratio", "size"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function Image(_ref) {
  var children = _ref.children,
      className = _ref.className,
      component = _ref.component,
      fullwidth = _ref.fullwidth,
      ratio = _ref.ratio,
      size = _ref.size,
      props = _objectWithoutProperties(_ref, _excluded);

  var Element = component; // Sizes

  var sizeMap = {
    '16x16': 'is-16x16',
    '24x24': 'is-24x24',
    '32x32': 'is-32x32',
    '48x48': 'is-48x48',
    '64x64': 'is-64x64',
    '96x96': 'is-96x96',
    '128x128': 'is-128x128'
  };
  var isSize = size && sizeMap[size]; // Ratios

  var ratioMap = {
    square: 'is-square',
    '1by1': 'is-1by1',
    '5by4': 'is-5by4',
    '4by3': 'is-4by3',
    '3by2': 'is-3by2',
    '5by3': 'is-5by3',
    '16by9': 'is-16by9',
    '2by1': 'is-2by1',
    '3by1': 'is-3by1',
    '4by5': 'is-4by5',
    '3by4': 'is-3by4',
    '2by3': 'is-2by3',
    '3by5': 'is-3by5',
    '9by16': 'is-9by16',
    '1by2': 'is-1by2',
    '1by3': 'is-1by3'
  };
  var isRatio = ratio && ratioMap[ratio];
  var classes = (0, _classnames["default"])('image', className, isSize, isRatio, {
    'is-fullwidth': fullwidth
  });
  return /*#__PURE__*/_react["default"].createElement(Element, _extends({
    className: classes
  }, props), children);
}

Image.Content = _ImageContent["default"];
Image.propTypes = {
  className: _propTypes["default"].string,
  children: _propTypes["default"].node,
  component: _propTypes["default"].elementType,
  size: _propTypes["default"].oneOf(['16x16', '24x24', '32x32', '48x48', '64x64', '96x96', '128x128']),
  ratio: _propTypes["default"].oneOf(['square', '1by1', '5by4', '4by3', '3by2', '5by3', '16by9', '2by1', '3by1', '4by5', '3by4', '2by3', '3by5', '9by16', '1by2', '1by3']),
  fullwidth: _propTypes["default"].bool
};
Image.defaultProps = {
  className: undefined,
  children: null,
  component: 'figure',
  size: undefined,
  ratio: undefined,
  fullwidth: false
};
