// react-bulma-companion/lib/Breadcrumb/Breadcrumb.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Breadcrumb;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _BreadcrumbList = _interopRequireDefault(require("./BreadcrumbList"));

var _BreadcrumbListItem = _interopRequireDefault(require("./BreadcrumbListItem"));

var _excluded = ["align", "children", "className", "component", "separator", "size"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function Breadcrumb(_ref) {
  var align = _ref.align,
      children = _ref.children,
      className = _ref.className,
      component = _ref.component,
      separator = _ref.separator,
      size = _ref.size,
      props = _objectWithoutProperties(_ref, _excluded);

  var Element = component; // Align

  var alignMap = {
    center: 'is-centered',
    right: 'is-right'
  };
  var isAlign = align && alignMap[align]; // Separators

  var separatorMap = {
    arrow: 'has-arrow-separator',
    bullet: 'has-bullet-separator',
    dot: 'has-dot-separator',
    succeeds: 'has-succeeds-separator'
  };
  var isSeparator = separator && separatorMap[separator]; // Size

  var sizeMap = {
    small: 'is-small',
    medium: 'is-medium',
    large: 'is-large'
  };
  var isSize = size && sizeMap[size];
  var classes = (0, _classnames["default"])('breadcrumb', className, isAlign, isSeparator, isSize);
  return /*#__PURE__*/_react["default"].createElement(Element, _extends({
    className: classes
  }, props), children);
}

Breadcrumb.List = _BreadcrumbList["default"];
Breadcrumb.ListItem = _BreadcrumbListItem["default"];
Breadcrumb.propTypes = {
  className: _propTypes["default"].string,
  children: _propTypes["default"].node,
  component: _propTypes["default"].elementType,
  align: _propTypes["default"].oneOf(['center', 'right']),
  separator: _propTypes["default"].oneOf(['arrow', 'bullet', 'dot', 'succeeds']),
  size: _propTypes["default"].oneOf(['small', 'medium', 'large'])
};
Breadcrumb.defaultProps = {
  className: undefined,
  children: null,
  component: 'nav',
  align: undefined,
  separator: undefined,
  size: undefined
};
