// react-bulma-companion/lib/Button/Button.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Button;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _excluded = ["active", "children", "className", "color", "component", "focused", "fullwidth", "hovered", "inverted", "light", "link", "loading", "outlined", "rounded", "selected", "size"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function Button(_ref) {
  var active = _ref.active,
      children = _ref.children,
      className = _ref.className,
      color = _ref.color,
      component = _ref.component,
      focused = _ref.focused,
      fullwidth = _ref.fullwidth,
      hovered = _ref.hovered,
      inverted = _ref.inverted,
      light = _ref.light,
      link = _ref.link,
      loading = _ref.loading,
      outlined = _ref.outlined,
      rounded = _ref.rounded,
      selected = _ref.selected,
      size = _ref.size,
      props = _objectWithoutProperties(_ref, _excluded);

  if (process.env.NODE_ENV === 'development' && link) {
    // eslint-disable-next-line
    console.warn('Using the `link` prop is deprecated. Use `component="a"` instead.');
  }

  var Element = link ? 'a' : component; // Colors

  var colorMap = {
    primary: 'is-primary',
    link: 'is-link',
    info: 'is-info',
    success: 'is-success',
    warning: 'is-warning',
    danger: 'is-danger',
    white: 'is-white',
    black: 'is-black',
    light: 'is-light',
    dark: 'is-dark',
    text: 'is-text',
    ghost: 'is-ghost'
  };
  var isColor = color && colorMap[color]; // Sizes

  var sizeMap = {
    small: 'is-small',
    normal: 'is-normal',
    medium: 'is-medium',
    large: 'is-large'
  };
  var isSize = size && sizeMap[size];
  var classes = (0, _classnames["default"])('button', className, isColor, isSize, {
    'is-light': light,
    'is-fullwidth': fullwidth,
    'is-outlined': outlined,
    'is-inverted': inverted,
    'is-rounded': rounded,
    'is-hovered': hovered,
    'is-focused': focused,
    'is-active': active,
    'is-selected': selected,
    'is-loading': loading,
    'is-static': props["static"]
  }); // eslint-disable-next-line

  delete props["static"];
  return /*#__PURE__*/_react["default"].createElement(Element, _extends({
    className: classes
  }, props), children);
}

Button.propTypes = {
  className: _propTypes["default"].string,
  children: _propTypes["default"].node,
  component: _propTypes["default"].elementType,
  type: _propTypes["default"].oneOf(['button', 'submit', 'reset']),

  /**
    Deprecation Warning: Use `component="a"`.
  */
  link: _propTypes["default"].bool,
  color: _propTypes["default"].oneOf(['primary', 'link', 'info', 'success', 'warning', 'danger', 'white', 'black', 'light', 'dark', 'text', 'ghost']),
  light: _propTypes["default"].bool,
  size: _propTypes["default"].oneOf(['small', 'normal', 'medium', 'large']),
  fullwidth: _propTypes["default"].bool,
  outlined: _propTypes["default"].bool,
  inverted: _propTypes["default"].bool,
  rounded: _propTypes["default"].bool,
  hovered: _propTypes["default"].bool,
  focused: _propTypes["default"].bool,
  active: _propTypes["default"].bool,
  selected: _propTypes["default"].bool,
  loading: _propTypes["default"].bool,
  "static": _propTypes["default"].bool,
  disabled: _propTypes["default"].bool
};
Button.defaultProps = {
  className: undefined,
  children: null,
  component: 'button',
  type: 'button',
  link: false,
  color: undefined,
  light: false,
  size: undefined,
  fullwidth: false,
  outlined: false,
  inverted: false,
  rounded: false,
  hovered: false,
  focused: false,
  active: false,
  selected: false,
  loading: false,
  "static": false,
  disabled: false
};
