// react-bulma-companion/lib/Pagination/Pagination.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Pagination;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _PaginationEllipsis = _interopRequireDefault(require("./PaginationEllipsis"));

var _PaginationLink = _interopRequireDefault(require("./PaginationLink"));

var _PaginationList = _interopRequireDefault(require("./PaginationList"));

var _PaginationListItem = _interopRequireDefault(require("./PaginationListItem"));

var _PaginationNext = _interopRequireDefault(require("./PaginationNext"));

var _PaginationPrevious = _interopRequireDefault(require("./PaginationPrevious"));

var _excluded = ["align", "children", "className", "component", "rounded", "size"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function Pagination(_ref) {
  var align = _ref.align,
      children = _ref.children,
      className = _ref.className,
      component = _ref.component,
      rounded = _ref.rounded,
      size = _ref.size,
      props = _objectWithoutProperties(_ref, _excluded);

  var Element = component;
  var sizeMap = {
    small: 'is-small',
    medium: 'is-medium',
    large: 'is-large'
  };
  var isSize = size && sizeMap[size];
  var alignMap = {
    center: 'is-centered',
    right: 'is-right'
  };
  var isAlign = align && alignMap[align];
  var classes = (0, _classnames["default"])('pagination', className, isSize, isAlign, {
    'is-rounded': rounded
  });
  return /*#__PURE__*/_react["default"].createElement(Element, _extends({
    role: "navigation",
    className: classes
  }, props), children);
}

Pagination.Ellipsis = _PaginationEllipsis["default"];
Pagination.Link = _PaginationLink["default"];
Pagination.List = _PaginationList["default"];
Pagination.ListItem = _PaginationListItem["default"];
Pagination.Next = _PaginationNext["default"];
Pagination.Previous = _PaginationPrevious["default"];
Pagination.propTypes = {
  className: _propTypes["default"].string,
  children: _propTypes["default"].node,
  component: _propTypes["default"].elementType,
  size: _propTypes["default"].oneOf(['small', 'medium', 'large']),
  align: _propTypes["default"].oneOf(['center', 'right']),
  rounded: _propTypes["default"].bool
};
Pagination.defaultProps = {
  className: undefined,
  children: null,
  component: 'nav',
  size: undefined,
  align: undefined,
  rounded: false
};
