// react-bulma-companion/lib/Hero/Hero.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Hero;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _HeroHead = _interopRequireDefault(require("./HeroHead"));

var _HeroBody = _interopRequireDefault(require("./HeroBody"));

var _HeroFoot = _interopRequireDefault(require("./HeroFoot"));

var _HeroVideo = _interopRequireDefault(require("./HeroVideo"));

var _HeroButtons = _interopRequireDefault(require("./HeroButtons"));

var _excluded = ["bold", "children", "className", "color", "component", "size"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function Hero(_ref) {
  var bold = _ref.bold,
      children = _ref.children,
      className = _ref.className,
      color = _ref.color,
      component = _ref.component,
      size = _ref.size,
      props = _objectWithoutProperties(_ref, _excluded);

  var Element = component; // Colors

  var colorMap = {
    primary: 'is-primary',
    link: 'is-link',
    info: 'is-info',
    success: 'is-success',
    warning: 'is-warning',
    danger: 'is-danger',
    white: 'is-white',
    black: 'is-black',
    light: 'is-light',
    dark: 'is-dark'
  };
  var isColor = color && colorMap[color]; // Sizes

  var sizeMap = {
    small: 'is-small',
    medium: 'is-medium',
    large: 'is-large',
    halfheight: 'is-halfheight',
    fullheight: 'is-fullheight',
    'fullheight-navbar': 'is-fullheight-with-navbar'
  };
  var isSize = size && sizeMap[size];
  var classes = (0, _classnames["default"])('hero', className, isColor, isSize, {
    'is-bold': bold
  });
  return /*#__PURE__*/_react["default"].createElement(Element, _extends({
    className: classes
  }, props), children);
}

Hero.Head = _HeroHead["default"];
Hero.Body = _HeroBody["default"];
Hero.Foot = _HeroFoot["default"];
Hero.Video = _HeroVideo["default"];
Hero.Buttons = _HeroButtons["default"];
Hero.propTypes = {
  className: _propTypes["default"].string,
  children: _propTypes["default"].node,
  component: _propTypes["default"].elementType,
  color: _propTypes["default"].oneOf(['primary', 'link', 'info', 'success', 'warning', 'danger', 'white', 'black', 'light', 'dark']),
  size: _propTypes["default"].oneOf(['small', 'medium', 'large', 'halfheight', 'fullheight', 'fullheight-navbar']),
  bold: _propTypes["default"].bool
};
Hero.defaultProps = {
  className: undefined,
  children: null,
  component: 'section',
  color: undefined,
  size: undefined,
  bold: false
};
