// react-bulma-companion/lib/Table/TableHeadCell/TableHeadCell.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = TableHeadCell;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _excluded = ["children", "className", "color", "component", "narrow", "selected", "vcentered"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function TableHeadCell(_ref) {
  var children = _ref.children,
      className = _ref.className,
      color = _ref.color,
      component = _ref.component,
      narrow = _ref.narrow,
      selected = _ref.selected,
      vcentered = _ref.vcentered,
      props = _objectWithoutProperties(_ref, _excluded);

  var Element = component; // Colors

  var colorMap = {
    primary: 'is-primary',
    link: 'is-link',
    info: 'is-info',
    success: 'is-success',
    warning: 'is-warning',
    danger: 'is-danger',
    white: 'is-white',
    black: 'is-black',
    light: 'is-light',
    dark: 'is-dark'
  };
  var isColor = color && colorMap[color];
  var classes = (0, _classnames["default"])(className, isColor, {
    'is-selected': selected,
    'is-narrow': narrow,
    'is-vcentered': vcentered
  });
  return /*#__PURE__*/_react["default"].createElement(Element, _extends({
    className: classes
  }, props), children);
}

TableHeadCell.propTypes = {
  className: _propTypes["default"].string,
  children: _propTypes["default"].node,
  component: _propTypes["default"].elementType,
  color: _propTypes["default"].oneOf(['primary', 'link', 'info', 'success', 'warning', 'danger', 'white', 'black', 'light', 'dark']),
  selected: _propTypes["default"].bool,
  narrow: _propTypes["default"].bool,
  vcentered: _propTypes["default"].bool
};
TableHeadCell.defaultProps = {
  className: undefined,
  children: null,
  component: 'th',
  color: undefined,
  selected: false,
  narrow: false,
  vcentered: false
};
