// react-bulma-companion/lib/Table/Table.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Table;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _TableHead = _interopRequireDefault(require("./TableHead"));

var _TableFoot = _interopRequireDefault(require("./TableFoot"));

var _TableBody = _interopRequireDefault(require("./TableBody"));

var _TableRow = _interopRequireDefault(require("./TableRow"));

var _TableHeadCell = _interopRequireDefault(require("./TableHeadCell"));

var _TableDataCell = _interopRequireDefault(require("./TableDataCell"));

var _TableContainer = _interopRequireDefault(require("./TableContainer"));

var _excluded = ["bordered", "children", "className", "component", "fullwidth", "hoverable", "narrow", "striped"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function Table(_ref) {
  var bordered = _ref.bordered,
      children = _ref.children,
      className = _ref.className,
      component = _ref.component,
      fullwidth = _ref.fullwidth,
      hoverable = _ref.hoverable,
      narrow = _ref.narrow,
      striped = _ref.striped,
      props = _objectWithoutProperties(_ref, _excluded);

  var Element = component;
  var classes = (0, _classnames["default"])('table', className, {
    'is-bordered': bordered,
    'is-striped': striped,
    'is-narrow': narrow,
    'is-hoverable': hoverable,
    'is-fullwidth': fullwidth
  });
  return /*#__PURE__*/_react["default"].createElement(Element, _extends({
    className: classes
  }, props), children);
}

Table.Head = _TableHead["default"];
Table.Foot = _TableFoot["default"];
Table.Body = _TableBody["default"];
Table.Row = _TableRow["default"];
Table.HeadCell = _TableHeadCell["default"];
Table.DataCell = _TableDataCell["default"];
Table.Container = _TableContainer["default"];
Table.propTypes = {
  className: _propTypes["default"].string,
  children: _propTypes["default"].node,
  component: _propTypes["default"].elementType,
  bordered: _propTypes["default"].bool,
  striped: _propTypes["default"].bool,
  narrow: _propTypes["default"].bool,
  hoverable: _propTypes["default"].bool,
  fullwidth: _propTypes["default"].bool
};
Table.defaultProps = {
  className: undefined,
  children: null,
  component: 'table',
  bordered: false,
  striped: false,
  narrow: false,
  hoverable: false,
  fullwidth: false
};
