// react-bulma-companion/lib/Select/Select.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Select;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _SelectContent = _interopRequireDefault(require("./SelectContent"));

var _SelectOption = _interopRequireDefault(require("./SelectOption"));

var _excluded = ["children", "className", "color", "component", "fullwidth", "loading", "rounded", "size", "multiple"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function Select(_ref) {
  var children = _ref.children,
      className = _ref.className,
      color = _ref.color,
      component = _ref.component,
      fullwidth = _ref.fullwidth,
      loading = _ref.loading,
      rounded = _ref.rounded,
      size = _ref.size,
      multiple = _ref.multiple,
      props = _objectWithoutProperties(_ref, _excluded);

  var Element = component; // Colors

  var colorMap = {
    primary: 'is-primary',
    link: 'is-link',
    info: 'is-info',
    success: 'is-success',
    warning: 'is-warning',
    danger: 'is-danger',
    white: 'is-white',
    black: 'is-black',
    light: 'is-light',
    dark: 'is-dark'
  };
  var isColor = color && colorMap[color]; // Sizes

  var sizeMap = {
    small: 'is-small',
    medium: 'is-medium',
    large: 'is-large'
  };
  var isSize = size && sizeMap[size];
  var classes = (0, _classnames["default"])('select', className, isColor, isSize, {
    'is-fullwidth': fullwidth,
    'is-multiple': multiple,
    'is-rounded': rounded,
    'is-loading': loading
  });
  return /*#__PURE__*/_react["default"].createElement(Element, _extends({
    className: classes
  }, props), children);
}

Select.Content = _SelectContent["default"];
Select.Option = _SelectOption["default"];
Select.propTypes = {
  className: _propTypes["default"].string,
  children: _propTypes["default"].node,
  component: _propTypes["default"].elementType,
  color: _propTypes["default"].oneOf(['primary', 'link', 'info', 'success', 'warning', 'danger', 'white', 'black', 'light', 'dark']),
  size: _propTypes["default"].oneOf(['small', 'medium', 'large']),
  fullwidth: _propTypes["default"].bool,
  multiple: _propTypes["default"].bool,
  rounded: _propTypes["default"].bool,
  loading: _propTypes["default"].bool
};
Select.defaultProps = {
  className: undefined,
  children: null,
  component: 'div',
  color: undefined,
  size: undefined,
  fullwidth: false,
  multiple: false,
  rounded: false,
  loading: false
};
