// react-bulma-companion/lib/Title/Title.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Title;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _excluded = ["children", "className", "component", "size", "spaced", "subtitle", "header"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function Title(_ref) {
  var children = _ref.children,
      className = _ref.className,
      component = _ref.component,
      size = _ref.size,
      spaced = _ref.spaced,
      subtitle = _ref.subtitle,
      header = _ref.header,
      props = _objectWithoutProperties(_ref, _excluded);

  if (process.env.NODE_ENV === 'development' && header) {
    // eslint-disable-next-line
    console.warn('Using the `header` prop is deprecated. Use the `component` prop instead.');
  }

  var isTitle = !subtitle ? 'title' : 'subtitle';
  var Element = component;

  if (header) {
    if (size === '1') {
      Element = 'h1';
    } else if (size === '2') {
      Element = 'h2';
    } else if (size === '3' || !subtitle && !size) {
      Element = 'h3';
    } else if (size === '4') {
      Element = 'h4';
    } else if (size === '5' || subtitle && !size) {
      Element = 'h5';
    } else if (size === '6') {
      Element = 'h6';
    }
  }

  var sizeMap = {
    1: 'is-1',
    2: 'is-2',
    3: 'is-3',
    4: 'is-4',
    5: 'is-5',
    6: 'is-6'
  };
  var isSize = size && sizeMap[size];
  var classes = (0, _classnames["default"])(isTitle, className, isSize, {
    'is-spaced': spaced
  });
  return /*#__PURE__*/_react["default"].createElement(Element, _extends({
    className: classes
  }, props), children);
}

Title.propTypes = {
  className: _propTypes["default"].string,
  children: _propTypes["default"].node,
  component: _propTypes["default"].elementType,
  subtitle: _propTypes["default"].bool,
  size: _propTypes["default"].oneOf(['1', '2', '3', '4', '5', '6']),
  spaced: _propTypes["default"].bool,

  /**
    Deprecation Warning: Use `component`.
  */
  header: _propTypes["default"].bool
};
Title.defaultProps = {
  className: undefined,
  children: null,
  component: 'p',
  subtitle: false,
  size: undefined,
  spaced: false,
  header: false
};
